### PR TITLE
[ fix #5245 ] reset stPragmaOptions when getStoredInterface fails

### LINKED ISSUE #5245: An infinite loop?
I just encountered the following message repeatedly, seemingly in an infinite loop:
```
Failed to validate just-loaded interface: options changed
   Checking Bisimilarity.Weak.Alternative.Classical ([…]).
```
Before that Agda had checked, or started checking, several files multiple times:
<details><summary>(Lots of output.)</summary>
<p>

```
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Alternative ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Alternative.Equational-reasoning-instances ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.General ([…]).
 Checking Similarity.CCS ([…]).
  Checking Bisimilarity.CCS ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
  Checking Similarity ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Similarity.Equational-reasoning-instances ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.CCS.Examples.Classical ([…]).
  Checking Bisimilarity.CCS.Classical ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Alternative ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Up-to.Counterexamples ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
  Checking Expansion ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Expansion.Equational-reasoning-instances ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Similarity.General ([…]).
 Checking Bisimilarity.Delay-monad ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Expansion ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Expansion.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Similarity.Weak ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Expansion ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
   Checking Expansion.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Expansion ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
  Checking Expansion ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Similarity ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Similarity.Weak.Equational-reasoning-instances ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Alternative.Up-to.Delay-monad ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Weak.Alternative ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Alternative.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.CCS.Examples ([…]).
  Checking Bisimilarity.CCS ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.General ([…]).
  Checking Similarity.CCS ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Up-to.Counterexamples ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Delay-monad ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.CCS.Examples ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Comparison ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Expansion ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
   Checking Expansion.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Similarity.Weak ([…]).
    Checking Bisimilarity.Weak ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
      Checking Bisimilarity ([…]).
       Checking Bisimilarity.General ([…]).
     Checking Bisimilarity.Equational-reasoning-instances ([…]).
      Checking Bisimilarity ([…]).
       Checking Bisimilarity.General ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
     Checking Expansion ([…]).
      Checking Bisimilarity ([…]).
       Checking Bisimilarity.General ([…]).
    Checking Expansion.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Weak ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Expansion ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Weak.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Expansion ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Up-to ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Expansion.CCS ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Up-to.Closure ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.CCS ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Delay-monad ([…]).
   Checking Bisimilarity.Delay-monad ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.CCS ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
  Checking Expansion.Delay-monad ([…]).
   Checking Bisimilarity.Delay-monad ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.CCS.Examples.Natural-numbers ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Up-to.CCS ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Up-to ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Expansion ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Similarity ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking README.Pointers-to-results-from-the-paper ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Comparison ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Up-to.CCS ([…]).
  Checking Bisimilarity.Weak.CCS ([…]).
   Checking Bisimilarity.CCS ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Expansion ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Expansion.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Similarity.Weak ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity.Equational-reasoning-instances ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Expansion ([…]).
     Checking Bisimilarity ([…]).
      Checking Bisimilarity.General ([…]).
   Checking Expansion.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Weak ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
   Checking Expansion ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
  Checking Bisimilarity.Weak.Equational-reasoning-instances ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
  Checking Expansion ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Up-to ([…]).
  Checking Bisimilarity.Weak ([…]).
   Checking Bisimilarity ([…]).
    Checking Bisimilarity.General ([…]).
 Checking Expansion.CCS ([…]).
  Checking Bisimilarity.CCS ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
   Checking Bisimilarity.Equational-reasoning-instances ([…]).
    Checking Bisimilarity ([…]).
     Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Up-to.Closure ([…]).
  Checking Bisimilarity ([…]).
   Checking Bisimilarity.General ([…]).
 Checking Bisimilarity.Weak.Alternative.Classical ([…]).
```

</p>
</details>

I don't know what was going on. I could fix the problem by removing the `_build` directory.

### PR DESCRIPTION
**Pull request by uncle-betty**
See the history in #5245 for details.

Technically, this issue is not specific to interaction mode. So I'm not entirely happy with the test being in `interaction`. However, only the interaction mode tests seem to offer the flexibility to script tests that are more complex than "here, compile this `.agda` file".

### CODE REVIEW

**Comment by nad** [comment]
> So I'm not entirely happy with the test being in `interaction`.

It's fine to place the test there.

File: test/interaction/Issue5245.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+cd Issue5245
+
+cp safe lib-a.agda-lib
+cp b/safe b/lib-b.agda-lib
+cp c/safe c/lib-c.agda-lib
+
+${AGDA} --interaction --no-default-libraries --library-file=libs <<EOF
+IOTCM "LibA.agda" None Indirect (Cmd_load "LibA.agda" ["--ignore-interfaces"])
+EOF
```
Comment (nad) [inline test/interaction/Issue5245.sh]: Presumably the following code would also work:
```
${AGDA} --no-default-libraries --library-file=libs --ignore-interfaces LibA.agda
```

Comment (uncle-betty) [reply test/interaction/Issue5245.sh]: Oh, funny that you would say this. Yes, it does. I actually used that first. But then I figured "when in `interaction`, do as everybody else does". When in Rome...

Happy to change it back, though, if you'd prefer that.

Comment (nad) [reply test/interaction/Issue5245.sh]: The shorter code might be a bit easier to maintain.

Comment (uncle-betty) [reply test/interaction/Issue5245.sh]: Sure. I just rebased the pull request onto current `master` and made the change.

**Comment by nad** [comment]
I guess this pull request can be merged.

**Comment by uncle-betty** [comment]
I've been meaning to ask this for a while: What's the protocol for this? Who's generally doing merges of pull requests?

**Comment by nad** [comment]
Those with sufficient permissions typically merge their own pull requests.

**Comment by uncle-betty** [comment]
OK, cool. Thanks. I see a *Merge pull request* button. Let me find out whether this means that I have sufficient permissions.

**Pull request merged**